### PR TITLE
Update Supported Networks on Get Started page + visual improvements to search

### PR DESCRIPTION
**Pull request by benface**
This PR makes the Supported Networks section of the docs landing page ("Get Started") use the same source of truth as the Supported Networks page/table, excluding testnets and in a grid format. There is a link to the Supported Networks page for more details (chain IDs, specific support in Studio / on hosted, etc.)

I've also sneaked in some visual improvements to the search functionality, notably by making the search modal fade in, replacing the DocSearch magnifying glass icon with ours (more consistent), and tweaking the colors and spacing (particularly on mobile).

https://github.com/graphprotocol/docs/assets/1059139/6188986c-e1d0-40fd-bad6-703690de1a21

### CODE REVIEW

File: website/src/IndexPage.tsx [hunk 1]
```diff
@@ -192,219 +196,30 @@ export function Products() {
 }
 
 export function SupportedNetworks() {
-  const { t } = useI18n()
+  const { supportedNetworks } = useData() as { supportedNetworks: Awaited<ReturnType<typeof getSupportedNetworks>> }
```
Comment (saihaj) [inline website/src/IndexPage.tsx:199]: is there no automatic type inference? 🤔

**Review by saihaj — APPROVED** [review]

File: website/src/IndexPage.tsx (hunk 1 above)
Comment (benface) [reply website/src/IndexPage.tsx:199]: Hmm, how would it know? I checked and `useData()` doesn't accept a generic type either.

Comment (saihaj) [reply website/src/IndexPage.tsx:199]: @dimaMachina maybe we can add support for generic?

Comment (dimaMachina) [reply website/src/IndexPage.tsx:199]: This hook will be removed in near Nextra 4 in favor of server components

**Pull request merged**